### PR TITLE
Fixed bug when chaining a chord with a group

### PR DESCRIPTION
**Pull request by Nusnus**
Reported at Issue #5958

### CODE REVIEW

**Comment by Nusnus** [comment]
I don't understand what is the linter crying about, can you help me check the logs @auvipy ? Maybe you'll see what I'm missing 😅

**Comment by auvipy** [comment]
no worries about the linter now, we will figure it out later

File: t/integration/test_canvas.py [hunk 1]
```diff
@@ -859,6 +859,154 @@ def before_task_publish_handler(sender=None, body=None, exchange=None, routing_k
         redis_connection = get_redis_connection()
         redis_connection.delete(redis_key)
 
+    def test_chaining_upgraded_chords_pure_groups(self, manager, subtests):
+        """ This test is built to reproduce the github issue https://github.com/celery/celery/issues/5958
```
Comment (auvipy) [inline t/integration/test_canvas.py:863]: the integration tests looks good so far.

**Review by auvipy — CHANGES_REQUESTED** [review]
ignore the lint errors that is unrelated to this PR. thanks for tackling this so fast bro <3

**Comment by Alexander3** [comment]
Thanks for fixing this

File: celery/canvas.py [hunk 2]
```diff
@@ -916,6 +916,10 @@ def __or__(self, other):
             if not tasks:
                 # If the chain is empty, return the group
                 return other
+            if isinstance(tasks[-1], chord):
+                # CHAIN [last item is chord] | GROUP -> chain with chord body.
```
[the remark this answers is not in the document]
Comment (Nusnus) [reply celery/canvas.py:920]: Done, added 4 new unit tests for all of the changes of this PR

File: t/integration/test_canvas.py (hunk 1 above)
Comment (Nusnus) [reply t/integration/test_canvas.py:863]: Thanks!
These tests cover:
1. Fix in chord.__or__.
2. Fix in chain.__or__.
3. Make sure the workaround from the last 2 years is not broken from this change.

**Comment by Nusnus** [comment]
I've replied to your comments @auvipy and I believe we're ready for merge (once the tests will finish).
Just that linter error ergggg... haha

P.S
Thanks for the kind words! 💪

**Review by auvipy — APPROVED** [review]

**Pull request merged**